ui-core/src/components/inputs/datePicker/CalendarPicker.tsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import cx from 'classnames';
+import { ChevronLeft, ChevronRight } from '@osrd-project/ui-icons';
+import Calendar from './Calendar';
+import { CalendarSlot } from './type';
+import useCalendarPicker from './useCalendarPicker';
+
+export type CalendarPickerPrivateProps = {
+  selectedSlot?: CalendarSlot;
+  onDayClick: (day: Date) => void;
+  modalPosition: {
+    top: number;
+    left: number;
+  };
+  calendarPickerRef: React.RefObject<HTMLDivElement>;
+};
+
+export type CalendarPickerPublicProps = {
+  initialDate?: Date;
+  numberOfMonths?: 1 | 2 | 3;
+  selectableSlot?: CalendarSlot;
+};
+
+export type CalendarPickerProps = CalendarPickerPrivateProps & CalendarPickerPublicProps;
+
+const CalendarPicker: React.FC<CalendarPickerProps> = ({
+  initialDate,
+  selectedSlot,
+  selectableSlot,
+  numberOfMonths = 1,
+  onDayClick,
+  modalPosition,
+  calendarPickerRef,
+}) => {
+  const {
+    displayedMonthsStartDates,
+    showNavigationBtn,
+    canGoToNextMonth,
+    canGoToPreviousMonth,
+    handleGoToNextMonth,
+    handleGoToPreviousMonth,
+  } = useCalendarPicker({
+    initialDate,
+    selectedSlot,
+    selectableSlot,
+    numberOfMonths,
+    onDayClick,
+  });
+
+  return (
+    <div ref={calendarPickerRef} className="calendar-picker" style={modalPosition}>
+      {showNavigationBtn && (
+        <span
+          className={cx('calendar-navigation-btn', 'previous', {
+            disabled: !canGoToPreviousMonth,
+          })}
+          onClick={handleGoToPreviousMonth}
+        >
+          <ChevronLeft size="lg" />
+        </span>
+      )}
+      <div className={cx('calendar-list', { 'navigation-btn-hidden': !showNavigationBtn })}>
+        {displayedMonthsStartDates.map((date, index) => {
```

Comment:
displayedMonthsStartDates contains only 1 date, what is it used for ? If it's only a date, we don't need the array and we should rename the const
```suggestion
        {displayedMonthsStartDates.map((date, index) => {
```

Reply:
`displayedMonthsStartDates` store the start dates of each calendar. Based on the value of `numberOfMonths`, we can have one, two or tree calendar displayed. So based on `numberOfMonths`, we can have one, two or three dates in `displayedMonthsStartDates`